> I'm still curious about how to reproduce the [warnings during compilation](https://github.com/AFLplusplus/AFLplusplus/commit/654f389e73c9fd5b7e141b33ea28ab0fdda3178f) caused by `add_lto_linker(aflcc);` and `add_lto_passes(aflcc);` when `-c` is present. Maybe I should try more versions of llvm 11+

Well, I just found a new bug might be introduced by commit df0638ab87a37f0a3604f5a95a4164153a3bd582, when testing LTO with LLVM 11:

```text
/usr/lib/llvm-11/bin/clang++ `llvm-config-11 --cxxflags` -fno-rtti -fno-exceptions -fPIC -O3 -funroll-loops -fPIC -Wall -g -I ./include/ -DVERSION=\"++4.10a\" -Wno-variadic-macros -Wno-deprecated-copy-with-dtor -DLLVM_MINOR=0 -DLLVM_MAJOR=11  -Wno-deprecated-declarations -Wl,-znodelete -Wno-writable-strings -fno-rtti -fPIC -std=c++14 -shared instrumentation/SanitizerCoverageLTO.so.cc -o SanitizerCoverageLTO.so `llvm-config-11 --ldflags`  instrumentation/afl-llvm-common.o
warning: unknown warning option '-Wno-deprecated-copy-with-dtor'; did you mean '-Wno-deprecated-copy-dtor'? [-Wunknown-warning-option]
instrumentation/SanitizerCoverageLTO.so.cc:1504:54: error: no member named 'getFixedValue' in 'llvm::TypeSize'; did you mean 'getFixedSize'?
  Array->setAlignment(Align(DL->getTypeStoreSize(Ty).getFixedValue()));
                                                     ^~~~~~~~~~~~~
                                                     getFixedSize
/usr/lib/llvm-11/include/llvm/Support/TypeSize.h:137:12: note: 'getFixedSize' declared here
  uint64_t getFixedSize() const {
           ^
1 warning and 1 error generated.
make[1]: *** [GNUmakefile.llvm:445: SanitizerCoverageLTO.so] Error 1
```

Clang version:
```text
$ clang-11 --version
Ubuntu clang version 11.1.0-++20211011094159+1fdec59bffc1-1~exp1~20211011214622.5
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

LLVM 12+ is not affected.

--- a/src/afl-cc.c
+++ b/src/afl-cc.c
@@ -2059,7 +2059,6 @@ void add_lto_passes(aflcc_state_t *aflcc) {
 #endif
 
   insert_param(aflcc, "-Wl,--allow-multiple-definition");
-  insert_param(aflcc, aflcc->lto_flag);
 
 }
 
@@ -2295,10 +2294,6 @@ param_st parse_misc_params(aflcc_state_t *aflcc, u8 *cur_argv, u8 scan) {
 
     SCAN_KEEP(aflcc->have_pic, 1);
 
-  } else if (cur_argv[0] != '-') {
-
-    SCAN_KEEP(aflcc->non_dash, 1);
-
   } else if (!strcmp(cur_argv, "-m32") ||
 
              !strcmp(cur_argv, "armv7a-linux-androideabi")) {
@@ -2373,6 +2368,14 @@ param_st parse_misc_params(aflcc_state_t *aflcc, u8 *cur_argv, u8 scan) {
       final_ = PARAM_DROP;
 
     }
+
+  } else if (cur_argv[0] != '-') {
+
+    /* It's a weak, loose pattern, with very different purpose
+     than others. We handle it at last, cautiously and robustly. */
+
+    if (scan && cur_argv[0] != '@')  // response file support
+      aflcc->non_dash = 1;
 
   }
 
@@ -2912,10 +2915,16 @@ static void edit_params(aflcc_state_t *aflcc, u32 argc, char **argv,
     //     use. insert_param(aflcc, "-flegacy-pass-manager");
     // #endif
 
-    if (aflcc->lto_mode && !aflcc->have_c) {
+    if (aflcc->lto_mode) {
 
-      add_lto_linker(aflcc);
-      add_lto_passes(aflcc);
+      insert_param(aflcc, aflcc->lto_flag);
+
+      if (!aflcc->have_c) {
+
+        add_lto_linker(aflcc);
+        add_lto_passes(aflcc);
+
+      }
 
     } else {
 

